persistence/module.go

```diff
@@ -117,45 +78,35 @@ func (m *persistenceModule) GetBus() modules.Bus {
 	return m.bus
 }
 
-func (m *persistenceModule) NewContext(height int64) (modules.PersistenceContext, error) {
-	persistenceContext := PostgresContext{
-		Height:       height,
-		PostgresDB:   m.postgresConn,
-		BlockStore:   m.blockStore,
-		ContextStore: kvstore.NewMemKVStore(),
+func (m *persistenceModule) NewRWContext(height int64) (modules.PersistenceRWContext, error) {
+	db, err := ConnectAndInitializeDatabase(m.postgresURL, m.nodeSchema)
```

Comment:
I think this might be the source of the issues I'm running into in the #128-fixes/improvements. 

Can we reuse the same DB connection from `Create` here?